src/ExpressionEvaluator/CSharp/Source/ResultProvider/CSharpFormatter.cs

```diff
@@ -47,6 +48,18 @@ internal override string TrimAndGetFormatSpecifiers(string expression, out ReadO
             return RemoveLeadingAndTrailingContent(expression, 0, expression.Length, IsWhitespace, ch => ch == ';' || IsWhitespace(ch));
         }
 
+        internal override string SimplifyCompilerGeneratedName(string name)
+        {
+            // If it's not compiler-generated, nothing to do here.
+            if (!GeneratedNameParser.TryParseGeneratedName(name, out _, out var openBracketOffset, out var closeBracketOffset))
+            {
+                return name;
+            }
+
+            var result = name.Substring(openBracketOffset + 1, closeBracketOffset - openBracketOffset - 1);
+            return result;
```

Comment:
I'd add `GeneratedNameParser.GetOriginalName(string metadataName)` that implements this and then here just call that helper.

Reply:
Also, this is not entirely correct in general. If `name` is a method name it might need extra handling. See [https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Synthesized/GeneratedNameParser.cs,86](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Synthesized/GeneratedNameParser.cs,86). 

We can either make this work for any compiler generated member, or we should have a two helpers (`GetOriginalFieldName` and `GetOriginalLocalVariableName`) corresponding to the uses cases the debugger currently needs.